### PR TITLE
ci: run custom benches in ci [WPB-10163]

### PR DESCRIPTION
**Pull request by SimonThormeyer**
# What's new in this PR
See PR title/corresponding issue

----
##### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      BENCHER_PROJECT: core-crypto-mmbtki3h
+      BENCHER_API_TOKEN: ${{ secrets.BENCHER_API_TOKEN }}
+      BENCH_PARAMS: "--bench commit --bench encryption --bench key_package --bench create_group -- --quick"  # Run everything except mls_proteus
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rust-lang/setup-rust-toolchain@v1  # this implicitly caches Rust tools and build artifacts
+      - name: Install bencher
+        uses: bencherdev/bencher@main
+      - name: Run benchmarks on PR
+        if: github.event.pull_request.head.repo.full_name == github.repository && github.event_name == 'pull_request'
```
Comment (istankovic) [inline .github/workflows/benchmarks.yml:24]: Why is this line needed?

Comment (istankovic) [reply .github/workflows/benchmarks.yml:24]: Ah, I think I see why -- for PRs, we want to use the previously-measured `develop` as the baseline, right?
And then the second case below is just for pushes to `develop`, to create that baseline measurement?

Comment (SimonThormeyer) [reply .github/workflows/benchmarks.yml:24]: Yep, correct!

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -100,11 +100,13 @@ jobs:
...
           rustflags: ''
-      - uses: taiki-e/install-action@nextest
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@nextest
       - name: "Test CoreCrypto xtask tooling"
         run: cargo nextest run --verbose --manifest-path xtask/Cargo.toml
       - name: "Test CoreCrypto benchmarks"
-        run: sh crypto/benches/scripts/test.sh
+        # Test only this because the others are executed in the benchmark workflow
+        run: cargo nextest run --bench mls_proteus
```
Comment (istankovic) [inline .github/workflows/rust.yml:109]: Why not move this also to the benchmark workflow?

Comment (SimonThormeyer) [reply .github/workflows/rust.yml:109]: Good point, will do

File: .github/workflows/benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+jobs:
+  benchmark_with_bencher:
+    name: Benchmark with Bencher
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    env:
+      BENCHER_PROJECT: core-crypto-mmbtki3h
+      BENCHER_API_TOKEN: ${{ secrets.BENCHER_API_TOKEN }}
+      BENCH_PARAMS: "--bench commit --bench encryption --bench key_package --bench create_group --bench mls_proteus -- --quick"  # Run everything except mls_proteus
```
Comment (istankovic) [inline .github/workflows/benchmarks.yml:17]: The comment needs to be removed now. :-)

**Review by istankovic — APPROVED** [review]
There's a comment that needs to be removed, but otherwise looks good to me.
